src/Network/Ethereum/Web3/Provider.hs

```diff
@@ -14,28 +14,34 @@ module Network.Ethereum.Web3.Provider where
 import           Control.Concurrent.Async    (Async, async)
 import           Control.Exception           (try)
 import           Control.Monad.IO.Class      (MonadIO (..))
+import           Data.Proxy
 import           Network.Ethereum.Web3.Types
+import           Network.HTTP.Client         (Manager, newManager)
+import           Network.HTTP.Client.TLS     (tlsManagerSettings)
+
+
+
+data RPCNode =
+  RPCNode { rpcUri :: String
```

Comment:
`ServerUri` is not used any more in `JsonRpc.hs`.
I think that could be moved in this module and be used instead of String here and in `makeRPCNode`.